request/load_balancer_reverse_proxy.rst

```diff
@@ -7,66 +7,45 @@ an AWS Elastic Load Balancer) or a reverse proxy (e.g. Varnish for
 via HTTPS, the client's port and the hostname being requested.
 
-Solution: trusted_proxies
--------------------------
+.. _request-set-trusted-proxies:
 
-This is no problem, but you *do* need to tell Symfony what is happening
-and which reverse proxy IP addresses will be doing this type of thing:
+Solution: setTrustedProxies()
+-----------------------------
 
-.. configuration-block::
+To fix this, you need to tell Symfony which reverse proxy IP addresses to trust
+and what headers your reverse proxy uses to send information:
 
-    .. code-block:: yaml
+.. code-block:: php
 
-        # app/config/config.yml
-        # ...
-        framework:
-            trusted_proxies:  [192.0.0.1, 10.0.0.0/8]
+    // web/app.php
 
-    .. code-block:: xml
+    // ...
+    $request = Request::createFromGlobals();
 
-        <!-- app/config/config.xml -->
-        <?xml version="1.0" encoding="UTF-8" ?>
-        <container xmlns="http://symfony.com/schema/dic/services"
-            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-            xmlns:framework="http://symfony.com/schema/dic/symfony"
-            xsi:schemaLocation="http://symfony.com/schema/dic/services
-                http://symfony.com/schema/dic/services/services-1.0.xsd
-                http://symfony.com/schema/dic/symfony http://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+    // tell Symfony about your revers proxy
+    Request::setTrustedProxies(
+        // the ip address (or range) of your proxy
+        ['192.0.0.1', '10.0.0.0/8'],
 
-            <framework:config trusted-proxies="192.0.0.1, 10.0.0.0/8">
-                <!-- ... -->
-            </framework:config>
-        </container>
+        // trust *all* "X-Forwarded-*" headers
+        Request::HEADER_X_FORWARDED_ALL
 
-    .. code-block:: php
+        // or, if your proxy instead uses the "Forwarded" header
+        // Request::HEADER_FORWARDED
+    );
 
-        // app/config/config.php
-        $container->loadFromExtension('framework', array(
-            'trusted_proxies' => array('192.0.0.1', '10.0.0.0/8'),
-        ));
-
-In this example, you're saying that your reverse proxy (or proxies) has
-the IP address ``192.0.0.1`` or matches the range of IP addresses that use
-the CIDR notation ``10.0.0.0/8``. For more details, see the
-:ref:`framework.trusted_proxies <reference-framework-trusted-proxies>` option.
-
-You are also saying that you trust that the proxy does not send conflicting
-headers, e.g. sending both ``X-Forwarded-For`` and ``Forwarded`` in the same
-request.
-
-That's it! Symfony will now look for the correct headers to get information
-like the client's IP address, host, port and whether the request is
-using HTTPS.
+The Request object has several ``Request::HEADER_*`` constants that control exactly
+*which* headers from your reverse proxy are trusted. The argument is a bit field,
+so you can also pass your own value (e.g. ``0b00110``).
```

Comment:
I'm not sure about this part: `... so you can also pass your own value (e.g. 0b00110)`  Don't you think that'd be super confusing and hard to maintain?